generator/internal/http_option_utils.cc

```diff
@@ -195,7 +195,52 @@ void SetHttpDerivedMethodVars(
   absl::visit(HttpInfoVisitor(method, method_vars), parsed_http_info);
 }
 
-// Request fields not appering in the path may not wind up as part of the json
+absl::optional<QueryParameterInfo> DetermineQueryParameterInfo(
+    google::protobuf::FieldDescriptor const& field) {
+  static auto* const kSupportedWellKnownValueTypes = [] {
+    auto foo = std::make_unique<
+        std::unordered_map<std::string, protobuf::FieldDescriptor::CppType>>();
+    foo->emplace("google.protobuf.BoolValue",
+                 protobuf::FieldDescriptor::CPPTYPE_BOOL);
+    foo->emplace("google.protobuf.DoubleValue",
+                 protobuf::FieldDescriptor::CPPTYPE_DOUBLE);
+    foo->emplace("google.protobuf.FloatValue",
+                 protobuf::FieldDescriptor::CPPTYPE_FLOAT);
+    foo->emplace("google.protobuf.Int32Value",
+                 protobuf::FieldDescriptor::CPPTYPE_INT32);
+    foo->emplace("google.protobuf.Int64Value",
+                 protobuf::FieldDescriptor::CPPTYPE_INT64);
+    foo->emplace("google.protobuf.StringValue",
+                 protobuf::FieldDescriptor::CPPTYPE_STRING);
+    foo->emplace("google.protobuf.UInt32Value",
+                 protobuf::FieldDescriptor::CPPTYPE_UINT32);
+    foo->emplace("google.protobuf.UInt64Value",
+                 protobuf::FieldDescriptor::CPPTYPE_UINT64);
+    return foo.release();
+  }();
+
+  absl::optional<QueryParameterInfo> param_info;
+  // Only attempt to make non-repeated, simple fields query parameters.
+  if (!field.is_repeated() && !field.options().deprecated()) {
+    if (field.cpp_type() != protobuf::FieldDescriptor::CPPTYPE_MESSAGE) {
+      param_info = QueryParameterInfo{
+          field.cpp_type(), absl::StrCat("request.", field.name(), "()"),
```

Comment:
Here and below, I think we might need to look up the accessor name. (in case the field is called something like `delete` or `namespace` and the accessor is `delete_()`).

I'd guess there are 0 such cases in practice, so we can defer.